cmds/boot/stboot/data.go

```diff
@@ -36,22 +37,33 @@ func findDataPartition() error {
 		return fmt.Errorf("filesystem unknown: %s", dataPartitionFSType)
 	}
 
-	devices, err := getBlockDevs()
-	if err != nil {
-		return fmt.Errorf("block devices: %v", err)
-	}
-	if len(devices) == 0 {
-		return fmt.Errorf("no non-loopback block devices found")
+	var devices []string
+	for {
+		devices, err = getBlockDevs()
+		if err != nil {
+			return fmt.Errorf("getting block devices failed with: %v", err)
+		}
+		if len(devices) == 0 {
+			if timeout == 0 {
+				return fmt.Errorf("no non-loopback block devices found")
+			}
+			debug("waiting for block devices to appear...\n")
+
+			timeout--
+			time.Sleep(time.Second)
```

Comment:
a second? ow. This is usually a firmware scenario, one process, multiple cores, burn the core. 10 ms? 1?